libs/vkd3d/resource.c

```diff
@@ -778,6 +782,16 @@ HRESULT vkd3d_get_image_allocation_info(struct d3d12_device *device,
     allocation_info->SizeInBytes = requirements.memoryRequirements.size;
     allocation_info->Alignment = requirements.memoryRequirements.alignment;
 
+    /* If we might create an image with VRS usage, need to also check memory requirements without VRS usage.
+     * VRS usage can depend on heap properties and this can affect compression, tile layouts, etc. */
+    if (create_info.image_info.usage & VK_IMAGE_USAGE_FRAGMENT_SHADING_RATE_ATTACHMENT_BIT_KHR)
+    {
+        create_info.image_info.usage &= ~VK_IMAGE_USAGE_FRAGMENT_SHADING_RATE_ATTACHMENT_BIT_KHR;
+        VK_CALL(vkGetDeviceImageMemoryRequirementsKHR(device->vk_device, &requirement_info, &requirements));
+        allocation_info->SizeInBytes = max(requirements.memoryRequirements.size, allocation_info->SizeInBytes);
+        allocation_info->Alignment = max(requirements.memoryRequirements.alignment, allocation_info->Alignment);
```

Comment:
We might need to realign the reported size to the new alignment as well in case the VRS alignment is smaller but size is larger than the old values.

Reply:
Not sure I follow we do max here for both cases. What are the potential returned values that can cause issues here?

Reply:
Say old alignment is 64k, old size is 64k, VRS alignment is 32k and VRS size is 96k, then we end up with new alignment 64k but new size 96k, but we'd need to return a size of 128k.

Probably won't ever happen in practice though.

Reply:
So the requirement is that returned size must align? Not sure where that requirement is stated.

Reply:
Turns out we're already doing this elsewhere anyway, so nevermind.